domain/port/service/service.go

```diff
@@ -98,6 +101,31 @@ func (s *Service) GetMachineOpenedPorts(ctx context.Context, machineUUID string)
 	return s.st.GetMachineOpenedPorts(ctx, machineUUID)
 }
 
+// GetMachineOpenedPortsAndSubnets returns the opened port ranges, including the
+// subnet CIDRs they're open to, for each endpoint, for each unit on a machine.
+//
+// NOTE(jack-w-shaw): This method is a stub implementation, that always returns
+// nil for the endpoint SubnetCIDRs
+// TODO: Once endpoint bindings have been implemented into a DQLite domain, implement
+// this method properly to return the subnet CIDRs each endpoint is open to.
+func (s *Service) GetMachineOpenedPortsAndSubnets(ctx context.Context, machineUUID string) (map[string]port.GroupedPortRangesOnSubnets, error) {
+	byUnitByEndpoint, err := s.st.GetMachineOpenedPorts(ctx, machineUUID)
+	if err != nil {
+		return nil, errors.Errorf("failed to get opened ports for machine %s: %w", machineUUID, err)
```

Comment:
Fix this up please. As noted prior, these flow up the call stack and materialise as "failed to... failed to... failed to..."